app/helpers/application_helper.rb

```diff
@@ -34,4 +34,24 @@ def decorated_session
       @_decorated_session ||= SessionDecorator.new
     end
   end
+
+  def sp_session
+    session[:sp]
+  end
+
+  def loa1_signup_context?
+    sp_session && current_user && !current_user.recovery_code.present?
+  end
+
+  def loa3_context?
+    sp_session && sp_session[:loa3] && current_user.recovery_code.present?
```

Comment:
> However, whether or not the user is attempting a sign up via LOA3 or LOA1, canceling at any point before the final verification screen is going to delete their account

Why? That's not what I'm understanding from issue 1333. If the user has already confirmed their email and 2FA phone during the LOA 3 signup process, they have an account. Once they enter IdV, canceling should only cancel IdV, not their entire account.